src/main/scala/com.snowplowanalytics.iglu/ctl/commands/TableMigrate.scala

```diff
@@ -0,0 +1,90 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.iglu.ctl.commands
+
+import java.nio.file.Path
+
+import cats.data._
+import cats.effect._
+import cats.implicits._
+import fs2.Stream
+import com.snowplowanalytics.iglu.core.{SchemaKey, SchemaMap}
+import com.snowplowanalytics.iglu.schemaddl.{StringUtils => SchemaDDLStringUtils}
+import com.snowplowanalytics.iglu.schemaddl.redshift.{Column => DDLColumn}
+import com.snowplowanalytics.iglu.ctl.Command.DbConfig
+import com.snowplowanalytics.iglu.ctl.commands.TableCheck.TableCheckResult
+import com.snowplowanalytics.iglu.ctl.commands.TableCheck.TableCheckResult._
+import com.snowplowanalytics.iglu.ctl.{Common, Failing, Result, Storage}
+
+object TableMigrate {
+
+  val EventFieldSeparator = "\\t"
+
+  /**
+    * Primary method of table-migrate command. Checks whether table of the given schema
+    * has expected structure which is determined with last and previous versions of
+    * the given schema and if it has not expected structure, creates necessary migration
+    * SQL statements
+    */
+  def process(resolver: Path,
+              schema: SchemaKey,
+              dbSchema: String,
+              outputS3Path: String,
+              awsRole: String,
+              awsRegion: String,
+              useGzip: Boolean,
+              dbConfig: DbConfig)(implicit cs: ContextShift[IO], t: Timer[IO]): Result = {
+    val stream = for {
+      resolvedDbConfig <- Stream.eval(Storage.resolveDbConfig(dbConfig))
+      storage <- Stream.resource(Storage.initialize[IO](resolvedDbConfig)).translate[IO, Failing](Common.liftIO)
+      tableCheckRes <- Stream.eval(TableCheck.tableCheckSingle(resolver, schema, storage, dbSchema))
+      res <- Stream.eval(getResult(tableCheckRes, outputS3Path, awsRole, awsRegion, useGzip))
+    } yield res
+
+    stream.compile.toList
+      .leftMap(e => NonEmptyList.of(e))
+  }
+
+  def getResult(tableCheckResult: TableCheckResult, outputS3Path: String, awsRole: String, awsRegion: String, useGzip: Boolean): Failing[String] = {
+    val res = tableCheckResult match {
+      case TableNotDeployed(_) => Common.Error.Message("Table not deployed").asLeft
+      case TableMatched(_) => Common.Error.Message("Table matched").asLeft
+      case TableUnmatched(schemaKey, _, expectedColumns) => createMigrationSQLStatements(expectedColumns, outputS3Path, schemaKey, awsRole, awsRegion, useGzip).asRight
+    }
+    EitherT.fromEither[IO](res)
+  }
+
+  def createMigrationSQLStatements(expectedColumns: List[DDLColumn], outputS3Path: String, schemaKey: SchemaKey, awsRole: String, awsRegion: String, useGzip: Boolean): String = {
```

Comment:
Camel case convention assumes that acronyms treated as usual words: `SqlStatements`, `DdlColumns`. Also I think this function's name is unnecessary long.